aws-s3-account-public-access-block/main.tf

```diff
@@ -0,0 +1,28 @@
+locals {
+  is_all  = var.restrict == "all"
+  is_new  = var.restrict == "new"
+  is_none = var.restrict == "none"
+
+
+
+  # These only affect new acls and policies by rejecting requests that contain them
+  block_public_acls   = ! local.is_none # all or new
+  block_public_policy = ! local.is_none # all or new
+
+  # These affect existing buckets, policies, and acls
+  ignore_public_acls      = local.is_all
+  restrict_public_buckets = local.is_all
+
+}
+
+
+# See https://docs.aws.amazon.com/AmazonS3/latest/API/API_control_PublicAccessBlockConfiguration.html for more info
+resource "aws_s3_account_public_access_block" "block" {
```

Comment:
⚠️ **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing version constraint for provider "aws" in "required_providers"